crypto/evp/evp_lib.c

```diff
@@ -637,14 +645,21 @@ int EVP_CIPHER_get_key_length(const EVP_CIPHER *cipher)
 
 int EVP_CIPHER_CTX_get_key_length(const EVP_CIPHER_CTX *ctx)
 {
-    int ok;
-    size_t v = ctx->key_len;
-    OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
-
-    params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_KEYLEN, &v);
-    ok = evp_do_ciph_ctx_getparams(ctx->cipher, ctx->algctx, params);
-
-    return ok != 0 ? (int)v : EVP_CTRL_RET_UNSUPPORTED;
+    /* Check to see if we've cached this value previously */
+    if (ctx->key_length < 0) {
+        size_t v = ctx->key_len;
```

Comment:
It's fairly confusing why do we need two ctx members to hold the key length. Perhaps we don't?

Reply:
I didn't notice that.  It's set to the EVP_CIPHER's key length and only updated by the EVP_CIPHER_CTX_set_key_length() call.  The set params path isn't accounted for.

I'll update tomorrow.

Reply:
The existing value is for legacy paths only.  A naïve change introduced lots of test errors.
This could get messy...